manifests/agent.pp

```diff
@@ -1,26 +1,45 @@
 class clamps::agent (
-  $nonroot_users = '2',
-  $master = $::servername,
-  $amqserver = $::servername,
-  $ca     = $::settings::ca_server,
-  $amqpass = file('/etc/puppetlabs/mcollective/credentials'),
+  $amqpass             = file('/etc/puppetlabs/mcollective/credentials'),
+  $amqserver           = [$::servername],
+  $ca                  = $::settings::ca_server,
+  $master              = $::servername,
+  $metrics_port        = 2003,
+  $metrics_server      = undef,
+  $nonroot_users       = '2',
+  $num_facts_per_agent = 500,
+  $shuffle_amq_servers = true,
 ) {
 
+  file { '/etc/puppetlabs/clamps':
+    ensure => directory
+  }
+
+  file { '/etc/puppetlabs/clamps/num_facts':
```

Comment:
So I see you manage this, however the fact above will be plugin synced and excute before this file exists which I think will cause a trace because of this:
```ruby 
File.read('/etc/puppetlabs/clamps/num_facts'
```
no?